### PR TITLE
Add silence cache in front of BoltDB

### PR DESCRIPTION
**Pull request by fabxc**
@beorn7 @mattbostock 

This addresses the next iteration of too-much-serialization issues by simply adding an in-memory cache.
For the future, where we'll be looking at thousands of silences, we'll want to index silences in a smarter way.

We were also decoding all silences twice by accident – oops.

### CODE REVIEW

**Comment by beorn7** [comment]
@fabxc On my review queue... Just busy with an offsite today, so latency is somewhat random.

File: provider/boltmem/boltmem.go [hunk 1]
```diff
@@ -272,19 +288,14 @@ func (s *Silences) All() ([]*types.Silence, error) {
...
 				return err
 			}
-			ms.ID = binary.BigEndian.Uint64(k)
-
-			if err := json.Unmarshal(v, &ms); err != nil {
-				return err
-			}
-
-			res = append(res, types.NewSilence(&ms))
+			// The ID is is duplicated in the value and always equal
```
Comment (juliusv) [inline provider/boltmem/boltmem.go]: "is is" -> "is"

**Comment by juliusv** [comment]
:+1: otherwise

**Comment by beorn7** [comment]
At least the deletion of silences has a race condition.

**Comment by fabxc** [comment]
Sure – totally skipped that yesterday, thanks.

**Comment by beorn7** [comment]
:+1:

**Pull request merged**